### PR TITLE
Refactored supported regions, large anonymity country codes and other country codes to use std::set

### PR DESCRIPTION
**Pull request by tmancey**
Resolves https://github.com/brave/brave-browser/issues/9205

## Submitter Checklist:

- [x] Submitted a [ticket](https://github.com/brave/brave-browser/issues) for my issue if one did not already exist.
- [x] Used Github [auto-closing keywords](https://help.github.com/articles/closing-issues-via-commit-messages/) in the commit message.
- [ ] Added/updated tests for this change (for new code or code which already has tests).
- Verified that these changes build without errors on
  - [ ] Android
  - [ ] iOS
  - [ ] Linux
  - [x] macOS
  - [ ] Windows
- Verified that these changes pass automated tests (unit, browser, security tests) on
  - [ ] iOS
  - [ ] Linux
  - [x] macOS
  - [ ] Windows
- [x] Verified that all lint errors/warnings are resolved (`npm run lint`)
- [x] Ran `git rebase master` (if needed).
- [x] Ran `git rebase -i` to squash commits (if needed).
- [x] Tagged reviewers and labelled the pull request as needed.
- [ ] Request a security/privacy review as needed.
- [x] Add appropriate QA labels (QA/Yes or QA/No) to include the closed issue in milestone
- [ ] Public documentation has been updated as necessary. For instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-compatibility-issues-with-tracking-protection
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide

## Test Plan:

- Confirm ads is enabled for supported regions
- Confirm ads is disabled for unsupported regions
- Confirm https://github.com/brave/brave-core/pull/5067 is working as expected

## Reviewer Checklist:

- [ ] New files have MPL-2.0 license header.
- [ ] Request a security/privacy review as needed.
- [ ] Adequate test coverage exists to prevent regressions 
- [ ] Verify test plan is specified in PR before merging to source

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on.
- [ ] All relevant documentation has been updated.

### CODE REVIEW

**Comment by tmancey** [comment]
Windows x64 failed CI, restarting

File: vendor/bat-native-ads/src/bat/ads/internal/supported_regions.h [hunk 1]
```diff
@@ -0,0 +1,325 @@
...
+#ifndef BAT_ADS_INTERNAL_SUPPORTED_REGIONS_H_
+#define BAT_ADS_INTERNAL_SUPPORTED_REGIONS_H_
+
+#include <map>
+#include <set>
+#include <string>
+
+namespace ads {
+
+const std::map<int, std::set<std::string>> kSupportedRegions = {
```
Comment (gdregalo) [inline vendor/bat-native-ads/src/bat/ads/internal/supported_regions.h:15]: Global `std::map` is prohibited by https://google.github.io/styleguide/cppguide.html#Static_and_Global_Variables.
Read "Common patterns" section for a solution: array of pairs.
constexpr std::array<std::pair<int, const std::array<const char*, M>, N>.
Keep the collection in sorted order and use a binary search algorithm.

File: vendor/bat-native-confirmations/src/bat/confirmations/internal/country_codes.h [hunk 2]
```diff
@@ -0,0 +1,208 @@
...
+
+#ifndef BAT_CONFIRMATIONS_INTERNAL_COUNTRY_CODES_H_
+#define BAT_CONFIRMATIONS_INTERNAL_COUNTRY_CODES_H_
+
+#include <set>
+#include <string>
+
+namespace confirmations {
+
+const std::set<std::string> kLargeAnonymityCountryCodes = {
```
Comment (gdregalo) [inline vendor/bat-native-confirmations/src/bat/confirmations/internal/country_codes.h:14]: Read above comment.
constexpr std::array<const char*, N> kLargeAnonymityCountryCodes?

Comment (tmancey) [reply vendor/bat-native-confirmations/src/bat/confirmations/internal/country_codes.h:14]: I raised https://github.com/brave/brave-browser/issues/9164 to fix this globally in Brave Ads

File: vendor/bat-native-ads/src/bat/ads/internal/supported_regions.h (hunk 1 above)
Comment (tmancey) [reply vendor/bat-native-ads/src/bat/ads/internal/supported_regions.h:15]: I raised https://github.com/brave/brave-browser/issues/9164 to fix this globally in Brave Ads

**Comment by tmancey** [comment]
CI passed at https://ci.brave.com/job/pr-brave-browser-issues-9205/2/ (so not sure why the PR does not reflect this...)

**Pull request merged**